instance/tests/js/instance_app_spec.js

```diff
@@ -75,99 +106,337 @@ describe('Instance app', function () {
                 expect($scope.loading).not.toBeTruthy();
                 $scope.updateInstanceList();
                 expect($scope.loading).toBeTruthy();
-                httpBackend.flush();
+                flushHttpBackend(); // Clear timeouts created by httpBackend
                 expect($scope.loading).not.toBeTruthy();
             });
+        });
 
-            it('updates the selected instance when updating the list', function() {
-                instanceDetail.domain = 'old.example.com';
-                $scope.select($scope.instanceList[0]);
-                httpBackend.flush();
-                expect($scope.selected.instance.domain).toEqual('old.example.com');
+        describe('swampdragon events', function() {
+            beforeEach(function() {
+                spyOn($scope, 'updateInstanceList');
+            });
+            it('update the instance list whenever an instance is updated', function() {
+                swampdragon.sendChannelMessage({type: "instance_update"});
+                expect($scope.updateInstanceList).toHaveBeenCalled();
+            });
+            it('update the instance list whenever a AppServer is updated', function() {
+                swampdragon.sendChannelMessage({type: "openedx_appserver_update"});
+                expect($scope.updateInstanceList).toHaveBeenCalled();
+            });
+            it('do not update the instance list for other changes', function() {
+                swampdragon.sendChannelMessage({type: "other_update"});
+                expect($scope.updateInstanceList).not.toHaveBeenCalled();
+            });
+        });
 
-                instanceDetail.domain = 'tmp.sandbox.opencraft.com';
-                $scope.updateInstanceList();
-                httpBackend.flush();
-                expect($scope.selected.instance.domain).toEqual('tmp.sandbox.opencraft.com');
+        describe('$scope.notify', function() {
+            it('can be called to show temporary notifications', function() {
+                expect($scope.notification).toBe(null);
+                $scope.notify('Update failed.', 'alert');
+                expect($scope.notification.message).toEqual('Update failed.');
+                expect($scope.notification.type).toEqual('alert');
+                $timeout.flush();
+                expect($scope.notification).toBe(null);
+            });
+        });
+    });
+
+
+    describe('Details controller', function() {
```

Comment:
@bradenmacdonald "Details" is a bit generic. Maybe go for "Instance Details controller" here? (If I understand correctly, you'd probably want/need to update the name of the actual controller too. I'm not sure how much work that would be, so feel free to skip this if it's not an easy update.)